joomla/joomla-cms
# [4.2] PHP8.0 fix for BaseDatabaseModel: Only variable references

Pull Request to fix on PHP 8.0 an issue with "Notice: Only variable references should be returned by reference in /srv/www/joomla/joomla-cms/libraries/src/MVC/Model/BaseDatabaseModel.php on line 418"

### Testing Instructions
Environment:
- **PHP Version: 8.0.17**
- Web Server: Apache/2.4.53 (Unix) OpenSSL/1.1.1n
- Joomla! Version: Joomla! 4.2.0-alpha2-dev Development [ Kuamini ] 29-March-2022 18:41 GMT

In a fresh installed Joomla 4 with the sample data for testing, under "Menus" open a menu item like "All Modules".
Or directly go to: administrator/index.php?option=com_menus&view=items&menutype=all-modules
(most of the other menu items have this issue)
and see the notice appear
> Notice: Only variable references should be returned by reference in /srv/www/joomla/joomla-cms/libraries/src/MVC/Model/BaseDatabaseModel.php on line 418

### Actual result BEFORE applying this Pull Request

> Notice: Only variable references should be returned by reference in /srv/www/joomla/joomla-cms/libraries/src/MVC/Model/BaseDatabaseModel.php on line 418

![before-pr](https://user-images.githubusercontent.com/1217850/165259729-3e5b6168-1fdf-4b87-a75f-6cb2cebe2c2e.png)

### Expected result AFTER applying this Pull Request

![after-pr](https://user-images.githubusercontent.com/1217850/165259725-40293930-242f-495a-9e79-9b6640107863.png)
